### ISSUE TITLE
Cannot go to definition into std lib when using modules

### ISSUE THREAD

**Issue by dryewo** [Microsoft/vscode-go#2145]
**TL;DR:** I cannot Go to Definition of builtin functions when working on projects that use go modules in directories outside GOPATH.
* It works for third-party functions from correctly imported packages.
* It works for builtin functions if a project is not using modules and is located under GOPATH

OS: OS X 10.12.6
VS Code version: 1.29.1
vscode-go version: 0.7.0
go version: 1.11.2
go env GOROOT: /usr/local/opt/go/libexec

Steps to reproduce:

1. Follow quick start steps to create a project: https://github.com/golang/go/wiki/Modules#quick-start-example (including `go build`) - pick a dir outside GOPATH, it's important.
2. Open VS code in the directory (using `code .`, for example), open _hello.go_ file.
3. Right-click on `sampler.Hello` and choose "Go to Definition" - it will open _sampler.go_ file. Close it.
4. Right-click on `fmt.Println` and choose "Go to Definition"
5. It shows error message:
    Unable to open 'print.go': File not found (file:///$GOROOT/src/fmt/print.go).

![image](https://user-images.githubusercontent.com/3134153/48981564-5042f380-f0d7-11e8-8aff-ec243a20b2ba.png)

I had one hypothesis, that it somehow does not shell expand the `$GOROOT`.
So, I tried as root:

```
mkdir -p '/$GOROOT/src/fmt/'
touch '/$GOROOT/src/fmt/print.go'
```

After this, "Go to Definition" on `fmt.Println` opens the blank file without showing an error message, which proves the hypothesis to be true.

Until this is fixed, is there maybe a workaround for this?

**Comment by dryewo** [comment]
Found a workaround:

```
$ sudo ln -sf `go env GOROOT` '/$GOROOT'
```

This will create a symbolic link in `/` with literal name `$GOROOT` (unexpanded by shell), so that VS Code can still open files using unexpanded paths like `/$GOROOT/src/fmt/print.go`.

**Comment by saibing** [comment]
@dryewo 
[bingo](https://github.com/saibing/bingo) is a new go language server that support go 1.11 module.

you can try it.

**Comment by awking** [comment]
I have the same problem,  so confusing

**Comment by ramya-rao-a** [comment]
Thanks for reporting @dryewo.

@ianthehat any ideas?

**Comment by justlaputa** [comment]
same problem, any suggestion?

**Comment by wilbertliu** [comment]
I'm experiencing similar thing. I'd just put the link here to avoid duplication. 👇🏻
https://github.com/Microsoft/vscode-go/issues/2149#issuecomment-442085770

**Comment by ianthehat** [comment]
This was fixed about a month ago, there is a function called posToString in godef that patches the $GOROOT on the way out.
Could you try running godef-mod by hand and seeing what it does?

**Comment by wilbertliu** [comment]
> This was fixed about a month ago, there is a function called posToString in godef that patches the $GOROOT on the way out.
> Could you try running godef-mod by hand and seeing what it does?

I'm not sure I understand @ianthehat. So I ran this `godef-gomod <some file>`, and this is what I got:
```
godef: Expressions not yet supported <some file>
```

Anything missed here?

**Comment by ianthehat** [comment]
If you pass a string to godef it assumes it is a go expression not a file name
You need to do something like

    godef-gomod -f godef.go -o 640

Where -f takes the filename and -o is the byte offset into the file of the cursor. There is a VS-Code extension that will tell you this number.

**Comment by dpromanko** [comment]
I'm having the same issue. I changed my Go: Docs Tool to gogetdoc for the time being.

**Comment by wilbertliu** [comment]
> If you pass a string to godef it assumes it is a go expression not a file name
> You need to do something like
> 
> ```
> godef-gomod -f godef.go -o 640
> ```
> Where -f takes the filename and -o is the byte offset into the file of the cursor. There is a VS-Code extension that will tell you this number.

I'm not sure I understand what I should do @ianthehat. But here's what I did and the result:
```
godef-gomod -f main.go -o 606
godef: Offset 606 was not a valid identifier
```

Yes @dpromanko, `gogetdoc` works well. But isn't that so slow?

**Comment by dpromanko** [comment]
Yes it is quite slow. But for now it works.

**Comment by ramya-rao-a** [comment]
I can confirm that the issue where `Go to definition` fails for symbols from the std library with the `File not found (file://$GOROOT/src...)` is fixed after updating to the latest of `godef-gomod` using the `Go: Install/Update Tools` command
